modules/effects/src/map_to_action.ts

```diff
@@ -0,0 +1,168 @@
+import { Action } from '@ngrx/store';
+import {
+  defer,
+  merge,
+  Notification,
+  Observable,
+  OperatorFunction,
+  Subject,
+  NotificationKind,
+} from 'rxjs';
+import {
+  concatMap,
+  dematerialize,
+  filter,
+  finalize,
+  map,
+  materialize,
+} from 'rxjs/operators';
+
+/** Represents config with named paratemeters for mapToAction */
+export interface MapToActionConfig<
+  Input,
+  OutputAction extends Action,
+  ErrorAction extends Action,
+  CompleteAction extends Action,
+  UnsubscribeAction extends Action
+> {
+  // Project function that produces the output actions in success cases
+  project: (action: Input, index: number) => Observable<OutputAction>;
+  // Error handle function for project
+  error: (error: any, action: Input) => ErrorAction;
+  // Optional complete action provider
+  // count is the number of actions project emitted before completion
+  // action is the action that was completed
+  complete?: (count: number, action: Input) => CompleteAction;
+  // Optional flattening operator
+  operator?: <InputAction, OutputAction>(
+    project: (input: InputAction, index: number) => Observable<OutputAction>
+  ) => OperatorFunction<InputAction, OutputAction>;
+  // Optional unsubscribe action provider
+  // count is the number of actions project emitted before unsubscribing
+  // action is the action that was unsubscribed from
+  unsubscribe?: (count: number, action: Input) => UnsubscribeAction;
+}
+
+/**
+ * Wraps project fn with error handling making it safe to use in Effects.
+ * Takes either config with named properties that represent different possible
+ * callbacks or project/error callbacks that are required.
+ */
+export function mapToAction<
+  Input,
+  OutputAction extends Action,
+  ErrorAction extends Action
+>(
+  project: (action: Input, index: number) => Observable<OutputAction>,
+  error: (error: any, action: Input) => ErrorAction
+): (source: Observable<Input>) => Observable<OutputAction | ErrorAction>;
+export function mapToAction<
+  InputAction extends Action,
```

Comment:
This should be `Input` and shouldn't extend `Action`

Reply:
done